documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,258 @@ Control characters are not allowed in a specification.
+
+**How to fix the violation**: Ensure that the response object has a property named `value` of `array` type.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2027" />R2027 DefaultMustBeInEnum
+**Output Message**: The default value is not one of the values enumerated as valid for this element.
+
+**Description**: The value assigned as a default for an enum property must be present in the enums' list.
+
+**Why the rule is important**: Generated SDKs in types languages may fail to compile if we try to enforce a default value that is not a part of the enums defined in the list and in other languages may fail in serialization/deserialization phases.
+
+**How to fix the violation**: Ensure that the default desired actually exists in the enums' list.
+
+**Bad Example**:
+```
+"status":{
+  "type": "string",
+  "enum": [
+    "Succeeded",
+    "Updating",
+    "Deleting",
+    "Failed"
+  ],
+  "default": "Terminated"
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2047" />R2047 NamePropertyDefinitionInParameter
+**Output Message**: Parameter Must have the "name" property defined with non-empty string as its value.
+
+**Description**: A parameter must have a `name` property for the SDK to be properly generated. 
+
+**Why the rule is important**: AutoRest fails to generate code if the `name` property is not provided for a parameter.
+
+**How to fix the violation**: Add a non-empty `name` property to the parameter.
+
+**Bad Example**:
+```
+"MyParam":{
+  "type": "string",
+  "in": "path",
+  "description": "sample param"
+}
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R1001" />R1001 OperationIdNounVerb
+**Output Message**: Per the Noun_Verb convention for Operation Ids, the noun '{0}' should not appear after the underscore.
+
+**Description**: OperationId should be of the form `Noun_Verb`. 
+
+**Why the rule is important**: AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the `Verb` becomes the name of the method in that class. Not adhering to this format can either cause AutoRest to fail or can generate semantically incorrect SDK.
```

Comment:
I'd probably add "Impact to generated code" in this rule and describe the above. Regarding why the rule is important, I would probably say something about grouping methods in a way that makes sense for the end consumer of the service.